source/agora/node/Validator.d

```diff
@@ -605,12 +605,16 @@ public class Validator : FullNode, API
         const avail_height = enrolled == ulong.max ?
                                 next_height : enrolled + this.params.ValidatorCycle;
 
-        const enrollment = this.enroll_man.createEnrollment(enroll_key, avail_height);
-        log.trace("Sending Enrollment for enrolling at height {} (to validate blocks {} to {})",
-            avail_height, avail_height + 1, avail_height + this.params.ValidatorCycle);
-        this.enroll_man.enroll_pool.addValidated(enrollment, avail_height);
-        this.network.peers.each!(p => p.client.sendEnrollment(enrollment));
-        return enrollment;
+        if (height + 2 >= avail_height) // We are near the end of the cycle
+        {
+            const enrollment = this.enroll_man.createEnrollment(enroll_key, avail_height);
+            log.trace("Sending Enrollment for enrolling {} at height {} (to validate blocks {} to {})",
+                this.enroll_man.getEnrollmentPublicKey(), avail_height, avail_height + 1, avail_height + this.params.ValidatorCycle);
+            this.enroll_man.enroll_pool.addValidated(enrollment, avail_height);
+            this.network.peers.each!(p => p.client.sendEnrollment(enrollment));
+            return enrollment;
+        }
+        return Enrollment.init;
```

Comment:
Ideally this function shouldn't return anything, but I think something in the test uses it.

Reply:
This is just a general message for later though.

Reply:
Yes some tests use the returned enrollment to check other nodes have it in their pools.